libtenzir/src/active_partition.cpp

```diff
@@ -475,16 +333,8 @@ active_partition_actor::behavior_type active_partition(
       caf::delayed_anon_send(caf::actor_cast<caf::actor>(self), 100ms, msg);
       return;
     }
-    TENZIR_VERBOSE("{} shuts down after persisting partition state", *self);
-    // TODO: We must actor_cast to caf::actor here because 'shutdown' operates
-    // on 'std::vector<caf::actor>' only. That should probably be generalized
-    // in the future.
-    auto indexers = std::vector<caf::actor>{};
-    indexers.reserve(self->state.indexers.size());
-    auto copy = std::exchange(self->state.indexers, {});
-    for ([[maybe_unused]] auto&& [qf, indexer] : std::move(copy))
-      indexers.push_back(caf::actor_cast<caf::actor>(std::move(indexer)));
-    shutdown<policy::parallel>(self, std::move(indexers));
+    TENZIR_DEBUG("{} shuts down after persisting partition state", *self);
+    self->quit();
```

Comment:
We should consider quitting with `msg.reason` here or ideally `diagnostic::error(msg.reason).note("further context").to_error()`.